### PR TITLE
DAS-38 : Facets doesn't work with the new Kryo integration

### PR DESCRIPTION
**Pull request by swsachith**
Adding facet support with comma separated Strings

### CODE REVIEW

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java [hunk 1]
```diff
@@ -49,28 +49,33 @@ public void testFacetUDF() throws AnalyticsException {
...
-		                        + "(tableName \"Log\","
-		                        + "schema \"log_level STRING, message STRING, tenant INTEGER, composite FACET -i\""
-		                        + ")");
+                "CREATE TEMPORARY TABLE Log USING CarbonAnalytics "
+                        + "OPTIONS"
+                        + "(tableName \"Log\","
+                        + " schema \"log_level STRING, message STRING, tenant INTEGER\""
+                        + ")");
+
+        ex.executeQuery(1,
```
Comment (chanikag) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java]: Code is not aligned.

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java [hunk 2]
```diff
@@ -49,28 +49,33 @@ public void testFacetUDF() throws AnalyticsException {
...
+                "CREATE TEMPORARY TABLE facetTest USING CarbonAnalytics "
+                        + "OPTIONS"
+                        + "(tableName \"facetTest\","
+                        + " schema \"log_level STRING, message STRING, tenant INTEGER, composite FACET -i\""
+                        + ")");
 
-		// for a single facet
-		AnalyticsQueryResult result = ex.executeQuery(1, "SELECT facet1(tenant) from Log");
-		Assert.assertEquals(result.getRows().size(), INFO_MESSAGES + ERROR_MESSAGES);
+        ex.executeQuery(1,"INSERT INTO TABLE facetTest SELECT log_level,message,tenant, facet2(tenant,log_level) from Log");
```
Comment (chanikag) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java]: Code is not aligned.

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java (hunk 1 above)
Comment (swsachith) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java]: fixed with the PR : https://github.com/wso2/carbon-analytics/pull/137

File: components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java (hunk 2 above)
Comment (swsachith) [inline components/analytics-processors/org.wso2.carbon.analytics.spark.core/src/test/java/org/wso2/carbon/analytics/spark/core/udf/FacetSparkTest.java]: Fixed with the PR : https://github.com/wso2/carbon-analytics/pull/137

File: components/analytics-core/org.wso2.carbon.analytics.datasource.core.test/src/main/java/org/wso2/carbon/analytics/datasource/core/AnalyticsRecordStoreTest.java [hunk 3]
```diff
@@ -108,12 +108,11 @@ private Record createRecord(int tenantId, String tableName, String serverName, S
...
         List<Record> result = new ArrayList<Record>();
         Map<String, Object> values;
-        List<String> afacet;
+        String afacet;
         long timeTmp;
         for (int j = 0; j < c; j++) {
             values = new HashMap<String, Object>();
-            afacet = new ArrayList<>();
-            afacet.addAll(Arrays.asList(new String[] {"SomeLocation", "SomeInnerLocation", "AVillage"}));
+            afacet = "SomeLocation,SomeInnerLocation,AVillage";
```
Comment (sinthuja) [inline components/analytics-core/org.wso2.carbon.analytics.datasource.core.test/src/main/java/org/wso2/carbon/analytics/datasource/core/AnalyticsRecordStoreTest.java]: variable name should be in camel case.

Comment (swsachith) [inline components/analytics-core/org.wso2.carbon.analytics.datasource.core.test/src/main/java/org/wso2/carbon/analytics/datasource/core/AnalyticsRecordStoreTest.java]: fixed with https://github.com/wso2/carbon-analytics/pull/137

**Pull request merged**